packages/x-components/src/components/modals/__tests__/base-modal.spec.ts

```diff
@@ -87,13 +93,57 @@ describe('testing Base Modal  component', () => {
 
     expect(wrapper.find(getDataTestSelector('default-slot-overridden')).exists()).toBe(true);
   });
+
+  it('changes the focus to the correct element when the modal opens', async () => {
+    let { wrapper, setOpen, appendToBody } = mountBaseModal({
+      // eslint-disable-next-line max-len
+      defaultSlot: `<div><button data-test="expected-focus">First button</button><button>Second button</button></div>`,
+      open: false
+    });
+
+    appendToBody();
+    await setOpen(true);
+
+    expect(wrapper.find(getDataTestSelector('expected-focus')).element).toBe(
+      document.activeElement
+    );
+
+    ({ wrapper, setOpen, appendToBody } = mountBaseModal({
+      // eslint-disable-next-line max-len
+      defaultSlot: `<div><button>First button</button><button tabindex="1" data-test="expected-focus">Second button</button></div>`,
+      open: false
+    }));
+
+    appendToBody();
+    await setOpen(true);
+
+    expect(wrapper.find(getDataTestSelector('expected-focus')).element).toBe(
+      document.activeElement
+    );
+  });
+
+  it("doesn't changes the focus if the focusOnOpen prop is false", async () => {
+    const { setOpen, appendToBody } = mountBaseModal({
+      defaultSlot: `<div><button>First button</button><button>Second button</button></div>`,
+      open: false,
+      focusOnOpen: false
+    });
+
+    appendToBody();
+    const focusedElementBeforeOpen = document.activeElement;
+    await setOpen(true);
+
```

Comment:
I think this would reinforce the test, as makes clear that setting `focusOnOpen: false` will prevail over the element with `tabindex="1"`
```suggestion
expect(wrapper.find(getDataTestSelector('expected-focus')).element).not.toBe(
      document.activeElement
    );
```

Reply:
I can see the adding `tabindex` adds to the clarity of the test, but I don't see this assertion being much useful.

We're already asserting that the focused element before and after opening the modal is the same one, so I find that adding another assertion checking if a different element is indeed not focused is redundant.